components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/validator/validators/V410Validator.java

```diff
@@ -245,4 +246,22 @@ public void validateApiResourceLevelAuthScheme() {
 
     }
 
+    @Override
+    public void validateApiDeployedGatewayType(GenericArtifact apiArtifact) {
+        log.info("Validating deployed gateway type for API {name: " + apiName + ", version: " + apiVersion
+                + ", provider: " + provider + "}");
+        try {
+            String environments = apiArtifact.getAttribute(APIConstants.API_OVERVIEW_ENVIRONMENTS);
+            if ("none".equals(environments)) {
+                log.warn("No gateway environments are configured for API {name: " + apiName + ", version: " + apiVersion
+                        + ", provider: " + provider + "}. Hence revision deployment will be skipped at migration");
+            } else {
+                log.info("Completed deployed gateway type validation for API {name: " + apiName + ", version: " + apiVersion
+                        + ", provider: " + provider + "}");
+            }
+        } catch (GovernanceException e) {
+            log.error("Error on retrieving API Gateway environment from API generic artifact");
```

Comment:
exception should be logged.
```suggestion
            log.error("Error on retrieving API Gateway environment from API generic artifact", e);
```